src/uct/ib/dc/dc_mlx5_devx.c

```diff
@@ -109,7 +109,13 @@ uct_dc_mlx5_iface_devx_dci_connect(uct_dc_mlx5_iface_t *iface,
     UCT_IB_MLX5DV_SET(qpc, qpc, log_msg_max, UCT_IB_MLX5_LOG_MAX_MSG_SIZE);
     UCT_IB_MLX5DV_SET(qpc, qpc, atomic_mode, UCT_IB_MLX5_ATOMIC_MODE);
     UCT_IB_MLX5DV_SET(qpc, qpc, rae, true);
-    UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.sl, iface->super.super.super.config.sl);
+    if (uct_ib_iface_is_roce(&iface->super.super.super)) {
+        UCT_IB_MLX5DV_SET(qpc, qpc, primary_address_path.sl,
+                          iface->super.super.super.config.sl << 4);
```

Comment:
PRM says that this field is "For IB port only" (`Table 1351 - INIT2RTR_QP Required Parameters`)
Do we need setting `eth_prio` instead as we do for RC QP?

Reply:
thanks! fixed